update store on updating attributes in cluster.go

il9ue opened the pull request:
Added logic that updates store with given attributes.

gyuho commented:
@il9ue Test fails and please `gofmt` your code. And please follow our contribution guide https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message.

Thanks!

il9ue commented:
@gyuho Changed few lines in proper logic and made sure passed test and gofmt! Sorry for few mistakes

Thanks!

gyuho commented on etcdserver/cluster.go:
```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
+			c.members[id].Attributes = attr
+			return true
+		}
 		plog.Panicf("error updating attributes of unknown member %s", id)
```
This line is not needed anymore, right?

gyuho commented:
This looks good, but defer to @heyitsanthony and @xiang90 for final reviews.

Thanks,

gyuho commented:
@il9ue  As explained in https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message, please change the PR and commit message to `etcdserver: update store on updating attributes` and squash two commits into one because they are doing one change for one file.

heyitsanthony commented on etcdserver/cluster.go:
```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
+	p := path.Join(memberStoreKey(id), attributesSuffix)
 	_, ok := c.removed[id]
 	if ok {
 		plog.Warningf("skipped updating attributes of removed member %s", id)
```
the if/else nesting can be helped a bit by `return false` after the `Warningf()`, then removing the following `else`

heyitsanthony commented on etcdserver/cluster.go:
```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
 	} else {
+		if _, err := c.store.Update(p, string(b), store.Permanent); err != nil {
+			plog.Panicf("update Attributes should never fail: %v", err)
+		} else {
```
`else` is unnecessary here too because of the panic

il9ue commented:
@gyuho updated the commit message and squashed two previous commits. Also removed lines after 
```
Panicf()
```
@heyitsanthony injected 
```
return false
```
just to clarify logic, thanks!

heyitsanthony commented on etcdserver/cluster.go:
```diff
@@ -334,13 +334,20 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
+		}
 	}
-	// TODO: update store in this function
 	return false
```
`return true` at the end since the update succeeded

heyitsanthony commented:
fixup the success return value and it's lgtm. thanks!

il9ue commented:
@heyitsanthony  injected ```return true``` for success. Thanks!

xiang90 commented:
@il9ue We actually update the attribute here https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L1044 now. We need to avoid this if you move the updating store to cluster.go

il9ue commented:
Will check why build failed, could you give me feedback to resolve errors? It passed when tested locally, Thanks!
